api/server/request_controller.go

```diff
@@ -0,0 +1,117 @@
+package server
+
+import (
+	"net/http"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/gin-gonic/gin"
+	"github.com/iron-io/functions/api"
+	"github.com/iron-io/runner/common"
+)
+
+type RequestController interface {
+	AppName() string
```

Comment:
Why really have a full app here? I mean, the full App struct has just name a config. The config is really necessary for this situation? 
This objects holds request required information and the only two information I think right now is required for the request is appName and routePath.
Make sense?